src/features/session_replay/aggregate/index.js

```diff
@@ -362,7 +363,8 @@ export class Aggregate extends AggregateBase {
           'rrweb.version': RRWEB_VERSION,
           'payload.type': recorderEvents.type,
           // customer-defined data should go last so that if it exceeds the query param padding limit it will be truncated instead of important attrs
-          ...(endUserId && { 'enduser.id': this.obfuscator.obfuscateString(endUserId) })
+          ...(endUserId && { 'enduser.id': this.obfuscator.obfuscateString(endUserId) }),
+          pageUrl: cleanURL(agentRuntime.origin)
```

Comment:
We might want to consider origin pageUrl vs "current" pageUrl, for SPA cases (not just here, but across the agent where we report it)

Reply:
@metal-messiah hey, I want to check my understanding... 

So I think the issue here is essentially that `agentRuntime` is  'fixed' after initialization (i totally missed that). Are we saying we might want to have a `currentUrl` available from `agentRuntime`? (so perhaps expose a method to update `model.currentUrl` and then we call it where needed?)

Reply:
Because if we dont, the ST/SR payload will always point at the origin, instead of the soft nav url, which could affect the ability to debug each payload (ie they look like they match urls but actually dont in reality due to soft navigations).

My question was more geared toward:
Should we report pageUrl (origin), as well as "currentUrl" (current spa route)

Reply:
- Seemingly no customer value here to pad extra data; solely for debugging?
- Does knowing the exact url on a replay chunk really help dbg? I'm not certain how so. But since you can get origin url by looking the associated PageView event, the current url if needed would be more helpful.
- The `PageAction` event's `pageUrl` should also be updated then to reflect current url (enhancement/feature change bump)

Reply:
pageAction already reports both the `pageUrl` (origin) and `currentUrl` (soft nav route)

Reply:
> Seemingly no customer value here to pad extra data; solely for debugging?

Definitely not, right now there are lots of joins that have to happen to get page view data to supplement replay data, that could be removed and rely on the payload as source of truth, which decreases load time in the UI and increases reliability of replays and traces.

Reply:
Then we should sync Logging's prop name, or value, no?

Reply:
> Does knowing the exact url on a replay chunk really help dbg? I'm not certain how so. But since you can get origin url by looking the associated PageView event, the current url if needed would be more helpful.

Yes because we often get cases where multiple replays are reporting simultaneously, and differentiating them becomes difficult. Think of the recent ST conversations where there are multiple payloads with the same PTID when we think it shouldnt be happening.  It would be nice to be able to rule out same page vs diff pages

Reply:
> Then we should sync Logging's prop name, or value, no?

Yeah we probably need to align all the payloads that report urls to report the same things

Reply:
Ok, I tickted the correction for logging in JIRA. Sounds like the actionable **here** is to change `pageUrl` & value to actually be `currentUrl` & corresponding value.

Reply:
(ok, sounds like we aren't asking for any change/addition to runtime obj right now)

 I prepped something locally on Friday and it was _pretty_ close.  I saw we were using the following in some places already, would this be acceptable? @cwli24 @metal-messiah 
`currentUrl: cleanURL('' + location)`